packages/yarnpkg-core/sources/Project.ts

```diff
@@ -1558,18 +1563,27 @@ export class Project {
     });
   }
 
+  /**
+   * The uncompressed install state currently persisted on disk.
+   */
+  private persistedInstallState: Buffer | null = null;
```

Comment:
This is similar to the `lockFileChecksum` check, for which we only store a checksum (even more relevant here since the uncompressed install state is rather large, and can reasonably still grow). I'd prefer to make the field public and move it alongside `lockFileChecksum` (and make it a checksum). Would the speed improvement hold?

Reply:
I tested with MD5 and SHA512 and they both take ~21ms to checksum a install state of 19.436588MB so I'll change it to use the checksum instead, I'm not quite sure why you'd want to make it public as it's just an internal optimization to avoid compressing the buffer

Reply:
Mostly consistency - I like similar things to go similar places and work the same way 😄 Having one thing public and the other private, as a reader without context, I'd certainly wonder why the visibility differs.

Reply:
Perhaps we should instead change `lockFileChecksum` to be private since the same thing applies to it

Reply:
Tested some other algorithms as well, SHA1 is the fastest on my machine
```
SHA1   - 8.91220000013709ms
SHA256 - 9.596099998801947ms
SHA512 - 21.658100001513958ms
MD5    - 23.582599997520447ms
```

Reply:
Yep; I picked SHA512 mainly to ensure that it could be used everywhere hashes were used (same reason why we use SHA512 to hash the descriptor / locators, even though we could use a lower hash). That, plus it avoids problems if what wasn't used in a cryptographic setting suddenly starts being so. If the cost is 10ms execution time, it sounds reasonable.

> Perhaps we should instead change lockFileChecksum to be private since the same thing applies to it.

I'd still prefer to keep it public. It may happen that the user would want to update the `lockfileChecksum` (for example if they write their own function to persist it).